app/src/main/java/io/homeassistant/companion/android/database/AppDatabase.kt

```diff
@@ -178,26 +223,55 @@ abstract class AppDatabase : RoomDatabase() {
-                            it.put("enabled", cursor.getInt(cursor.getColumnIndex("enabled")))
-                            it.put("registered", cursor.getInt(cursor.getColumnIndex("registered")))
-                            it.put("state", "")
-                            it.put("last_sent_state", "")
-                            it.put("state_type", "")
-                            it.put("type", "")
-                            it.put("icon", "")
-                            it.put("name", "")
-                        })
+                var migrationSuccessful = false
+                if (cursor.moveToFirst()) {
+                    try {
+                        while (cursor.moveToNext()) {
+                            sensors.add(ContentValues().also {
+                                it.put("id", cursor.getString(cursor.getColumnIndex("id")))
+                                it.put("enabled", cursor.getInt(cursor.getColumnIndex("enabled")))
+                                it.put(
+                                    "registered",
+                                    cursor.getInt(cursor.getColumnIndex("registered"))
+                                )
+                                it.put("state", "")
+                                it.put("last_sent_state", "")
+                                it.put("state_type", "")
+                                it.put("type", "")
+                                it.put("icon", "")
+                                it.put("name", "")
+                            })
+                        }
+                        migrationSuccessful = true
+                    } catch (e: Exception) {
+                        Log.e(TAG, "Unable to migrate, proceeding with recreating the table", e)
+                        createNotificationChannel()
+                        val notification = NotificationCompat.Builder(appContext, channelId)
+                            .setSmallIcon(R.drawable.ic_stat_ic_notification)
+                            .setContentTitle(appContext.getString(R.string.database_migration_failed))
+                            .setPriority(NotificationCompat.PRIORITY_HIGH)
+                            .build()
+                        with(NotificationManagerCompat.from(appContext)) {
+                            notify(NOTIFICATION_ID, notification)
+                        }
+                        runBlocking {
+                            try {
+                                integrationRepository.fireEvent("mobile_app.migration_failed", mapOf())
+                                Log.d(TAG, "Event sent to Home Assistant")
+                            } catch (e: Exception) {
+                                Log.e(TAG, "Unable to send event to Home Assistant", e)
+                                Toast.makeText(appContext, R.string.database_event_failure, Toast.LENGTH_LONG).show()
+                            }
+                        }
```

Comment:
By catching the exception, the migration won't throw the exception.  That means the the migration technically won't fail and be recreated.  Also it's duplicate code from above.